app/models/user/UserService.scala

```diff
@@ -368,6 +372,48 @@ class UserService @Inject()(conf: WkConf,
     }
   }
 
+  def publicWritesCompact(user: User, requestingUser: User, userCompactInfo: UserCompactInfo): Fox[JsObject] =
+    for {
+      _ <- Fox.successful(())
+      teamsJson = parseArrayLiteral(userCompactInfo.team_ids).indices.map(
+        idx =>
+          Json.obj(
+            "id" -> parseArrayLiteral(userCompactInfo.team_ids)(idx),
```

Comment:
Could the parseArrayLiteral calls be moved into `GetResultUserCompactInfo`?

Reply:
Not sure how. The << thing does not seem to be very flexible, not allowing me to just store it somewhere. What would be possible is to create another "raw" case class for storing all the array literal and returning a "parsed" case class so the service does not use parseArrayLiteral

Reply:
I see. Then let’s keep it like you did it for now (just one case class) but I’d ask to rename the fields to make this obvious for the usages. `userCompactInfo.teamIdsAsNullableArrayLiteral` or something like that. What do you think?

Later we can try to experiment if we can get implicit conversions in there, similar to GetInstant and GetObjectId from SqlTypeImplicits.scala

Reply:
Agree, though it shouldn't be nullable anymore with the array remove stuff parseArrayLiteral